DataPreparation/SOCATCPExport/export.py

```diff
@@ -50,12 +50,18 @@ def main():
     logging.info(f'Processing {level} file: {filename}')
     citation = ''
     L0_hashsums = []
+    is_next_version = None
     print(filepath)
 
+    if export_date is not None:
+      is_next_version = hashsum
+
     if level == 'L2':
       #fetch citation
       c.execute("SELECT citation, doi from citation where expocode = ? ",[expocode])
-      citation = ', doi:'.join(c.fetchone()).strip('\n').strip(', doi:10.1594/PANGAEA.xxxxxx')
+      citation = ', doi:'.join(c.fetchone()).strip('\n')
+      if (citation.find(', doi:10.1594/PANGAEA.xxxxxx'))>0:
+        citation = citation[0:citation.find(', doi:10.1594/PANGAEA.xxxxxx')]
```

Comment:
Code repetition: Running the find call twice isn't a performance problem because it's such a small task. However, it does mean that if it ever needs changing then it has to be changed twice which is a risk for future bugs. Better to assign the find index to a variable and use that.